File: wheeled_vehicle_models/18dof/pydof18_demos/HMMWV/demo_hmmwv_sundials.py
# Demo for hmmwv simulation using the half implicit integrator in python

import numpy as np
import pydof18 as dof18
import sys

# Input file
driver_file = "../../data/input/" + str(sys.argv[1]) + ".txt"

# Vehicle specification
vehParamsJson = "../../data/json/HMMWV/vehicle.json"
tireParamsJson = "../../data/json/HMMWV/tmeasy.json"


# Construct the sundials solver
solver = dof18.d18SolverSundials()
solver.Construct(vehParamsJson, tireParamsJson, driver_file)


# Set optional inputs
# Solver tolerance
solver.SetTolerances(1e-5, 1e-3)
# Set the maximum time step solver can take
solver.SetMaxStep(1e-2)

# Iniitial conditions
veh_st = dof18.VehicleState()
tirelf_st = dof18.TMeasyState()
tirerf_st = dof18.TMeasyState()
tirelr_st = dof18.TMeasyState()
tirerr_st = dof18.TMeasyState()
solver.Initialize(veh_st, tirelf_st, tirerf_st, tirelr_st, tirerr_st)

# Enable simulation output - sundials solver for now only outputs at 100 Hz
solver.SetOutput("../../data/output/" +
                 str(sys.argv[1]) + "_hmmwv18SundialsPy.csv")
solver.SetVerbose(False)
# Solve
solver.Solve(False)  # No FSA

File: wheeled_vehicle_models/11dof/demos/HMMWV/demo_hmmwv_sundials.cpp
#include <numeric>
#include <algorithm>
#include <iterator>

#include "dof11_sundials.h"

using namespace d11;

int main(int argc, char** argv) {
    // Driver inputs and reference trajectory
    std::string driver_file = "../data/input/" + std::string(argv[1]) + ".txt";

    // Vehicle specification
    std::string vehParamsJSON = (char*)"../data/json/HMMWV/vehicle.json";
    std::string tireParamsJSON = (char*)"../data/json/HMMWV/tmeasy.json";

    // Construct the solver
    d11SolverSundials solver;
    solver.Construct(vehParamsJSON, tireParamsJSON, driver_file);

    // Set optional inputs
    solver.SetTolerances(1e-5, 1e-3);
    solver.SetMaxStep(1e-2);

    // Initialize solver (set initial conditions)
    VehicleState veh_st;
    TMeasyState tiref_st;
    TMeasyState tirer_st;

    solver.Initialize(veh_st, tiref_st, tirer_st);

    // Enable output
    solver.SetOutput("../data/output/" + std::string(argv[1]) + "_hmmwv18Sundials.csv");
    solver.SetVerbose(false);

    // Solve without quadratures (no reference path required)
    solver.Solve(false);
}
